impress-org/givewp
# Feature: Payment Information Banner

## Description

This PR includes the functionality to manage, update and display the Payment information banner for the donation detail page.

Since the same Window object is being used for the ListTable and the AdminPage adjustments to the DonationsListTable were necessary to put this together. The Window declaration for "GiveDonations" has been moved outside of the ListTable component into a common space for both the Table and the Admin page. The types have also been adjusted throughout the ListTable components to reflect the conditional options being provided per display .

The payment banner is made up of a Legend component that handles the display of the title + the status field and 5 other editable fields that are all maintained with react-hook-form.  Users are now able to select a donation from the table and update the details associated with that donation. The data populated into the payment banner reflects the donations current live data.

## Fields 

- Amount: manages the current donations amount with a modal that allows you to set the amount.
- Fee Recovered: manages the current donations fee amount with a modal that allows you to set the amount.
- Forms: manages a list of donation forms that the donation is assigned to with a searchable dropdown.
- DatePicker: manages the selected date that the donation was made with a calendar component from react-dates.
- TimePicker: manages the time of the donation with regular inputs.
- DonationMethod: A static UI display of the donation method based on gateway.
- Status: manages the donation status with a provided list of options to a react-select dropdown component.


## Affects

Donations ListTable

Donation Admin Page

## Visuals

https://user-images.githubusercontent.com/75056371/229931143-54881592-65db-4a12-9fd0-fae63ac1d25e.mov

## Testing Instructions

- Select any donation from the Donation table.
- Edit the various donation fields and save the changes.
- Refresh the page or head back to the Donation table to verify the changes were saved.

## Pre-review Checklist
